### PR TITLE
fix Directory not empty error

### PR DESCRIPTION
**Pull request by pradeepbbl**
https://pulp.plan.io/issues/8595

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/tasking/storage.py [hunk 1]
```diff
@@ -60,6 +60,9 @@ def _delete(self):
             shutil.rmtree(self.path)
         except FileNotFoundError:
             pass
+        except OSError:
```
Comment (ipanova) [inline pulpcore/tasking/storage.py:63]: instead of having another except block can the line 60 be updated with `ignore_errors=True`?

**Comment by daviddavis** [comment]
Thanks for the patch. Can you add a changelog (ie `CHANGES/8595.bugfix`) to this?

**Comment by ipanova** [comment]
can the commit message be updated so it contains `closes #8595`?

File: pulpcore/tasking/storage.py (hunk 1 above)
Comment (pradeepbbl) [reply pulpcore/tasking/storage.py:63]: sure, no problem.

File: CHANGES/8595.bugfix [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Fixed issue with `shutil.rmtree`(https://github.com/pulp/pulpcore/blob/master/pulpcore/tasking/storage.py#L60)
+sometimes failing with 'OSError: [Errno 39] Directory not empty',
+fixed by adding `ignore_errors=True` ref: https://docs.python.org/3/library/shutil.html#shutil.rmtree
+
+For more details refer: https://pulp.plan.io/issues/8595
```
Comment (daviddavis) [inline CHANGES/8595.bugfix:5]: Probably too much technical detail for the changelog. Also, it'll automatically link the changelog entry to the issue (8595). How about this:

```suggestion
Fixed bug where publications sometimes fail with the error '[Errno 39] Directory not empty'.
```

**Comment by daviddavis** [comment]
One suggestion on the changelog. Also, can you squash your commits?

Other than that, this looks good to me. :+1:

**Review by daviddavis — APPROVED** [review]

**Review by ipanova — APPROVED** [review]

**Pull request merged**